pkg/sql/resolver.go

```diff
@@ -194,45 +207,49 @@ func (p *planner) ResolveTypeByID(id uint32) (*types.T, error) {
 	return nil, errors.Newf("type id reference @%d not allowed in this context", id)
 }
 
+// hydrateTypesInTableDescriptor uses typeLookup to install metadata in the
+// types present in a table descriptor. typeLookup retrieves the fully
+// qualified name and descriptor for a particular ID.
+func hydrateTypesInTableDescriptor(
+	desc *TableDescriptor, typeLookup func(id sqlbase.ID) (*tree.TypeName, *TypeDescriptor, error),
+) error {
+	for i := range desc.Columns {
+		col := &desc.Columns[i]
+		if col.Type.UserDefined() {
+			// Look up its type descriptor.
+			name, typDesc, err := typeLookup(sqlbase.ID(col.Type.StableTypeID()))
```

Comment:
ditto about defensiveness in this method

Reply:
I don't think this one needs to worry about it -- whoever calls this method and passes in the type lookup function should know what to do with the error.